src/main/java/pl/project13/maven/git/GitCommitIdMojo.java

```diff
@@ -472,6 +476,16 @@ void loadBuildTimeData(@NotNull Properties properties) {
     put(properties, BUILD_TIME, smf.format(buildDate));
   }
 
+  void loadBuildHostData(@NotNull Properties properties) {
+    String buildHost = null;
+    try {
+      buildHost = InetAddress.getLocalHost().getHostName();
+    } catch (UnknownHostException e) {
+      log("Unable to get build host");
```

Comment:
IP address is not exposed anymore, only host name.

From: Konrad Malawski [mailto:notifications@github.com]
Sent: 6 квітня 2015 р. 22:39
To: ktoso/maven-git-commit-id-plugin
Cc: Sergey Podolsky
Subject: Re: [maven-git-commit-id-plugin] Add git.build.host property (#178)


In src/main/java/pl/project13/maven/git/GitCommitIdMojo.java<https://github.com/ktoso/maven-git-commit-id-plugin/pull/178#discussion_r27823145>:

> @@ -472,6 +476,16 @@ void loadBuildTimeData(@NotNull Properties properties) {

>      put(properties, BUILD_TIME, smf.format(buildDate));

>    }

>

> +  void loadBuildHostData(@NotNull Properties properties) {

> +    String buildHost = null;

> +    try {

> +      buildHost = InetAddress.getLocalHost().getHostName();

> +    } catch (UnknownHostException e) {

> +      log("Unable to get build host");
Oh OK. I agree with having unknown as the value. Sounds good. I still do not want to expose any IP addresses unless explicitly turned on though. Some of the companies using this plugin are very sensitive about this kind of stuff.
…
On Apr 6, 2015 9:30 PM, "Sergey Podolsky" <notifications@github.com<mailto:notifications@github.com>> wrote: In src/main/java/pl/project13/maven/git/GitCommitIdMojo.java <https://github.com/ktoso/maven-git-commit-id-plugin/pull/178#discussion_r27822565> : > @@ -472,6 +476,16 @@ void loadBuildTimeData(@NotNull<https://github.com/NotNull> Properties properties) { > put(properties, BUILD_TIME, smf.format(buildDate)); > } > > + void loadBuildHostData(@NotNull<https://github.com/NotNull> Properties properties) { > + String buildHost = null; > + try { > + buildHost = InetAddress.getLocalHost().getHostName(); > + } catch (UnknownHostException e) { > + log("Unable to get build host"); I have updated the error message to be more verbose. And indeed don't set it if it's null. We have to put null value explicitly, because null is processed into "Unknown" later. I think it would be better to have "Unknown" value in the resulting property rather than omitting the property completely from the list of properties. Someone may get confused in case if property does not exist for some non-obvious reason. — Reply to this email directly or view it on GitHub <https://github.com/ktoso/maven-git-commit-id-plugin/pull/178/files#r27822565> .

—
Reply to this email directly or view it on GitHub<https://github.com/ktoso/maven-git-commit-id-plugin/pull/178/files#r27823145>.